### DISCUSSION TITLE
API function to load a cart from the internet

### DISCUSSION THREAD

**Discussion by Tele256** [nesbox/TIC-80#1362] [Ideas]
It would be very interesting if there would be an API function that loads and runs a cart from a link. So you would input the link from the tic80 website of the cart you want to load, and then when that code runs it will unload the current cart and download and load the cart from that link.
That way, people who have uploaded carts to the tic80 website could make a launcher-like app, where other people can download and play their games easily. You could also use this to make a bigger game than TIC-80 supports: You could have part 1 of your game, then when it ends you could load part 2 of the game without having the player do anything on their own.
It would certainly be a very nice feature.

**Comment by msx80** [comment]
sounds a bit overengineered for a fantasy console :)

**Reply by Tele256** [reply]
Maybe exclude the internet part, and have it load a cart from those of which you have on your machine. So it would be a load function. That would work pretty well too!

**Comment by RobLoach** [comment]
Alternatively, you could likely compress each cart into a binary chunk, and then load each load the cart data in from memory. Seems very hacky, but could potentially work.

**Reply by joshgoebel** [reply]
Wait where are these binary chunks stored?  Not sure how compressing into binary chunks == loading from internet?

**Reply by RobLoach** [reply]
Not sure your could store the binary chunk in a lua script :shrug: .... But, you could theoretically inject incrementally into the RAM with memset https://github.com/nesbox/TIC-80/wiki/RAM

**Reply by joshgoebel** [reply]
I see that as more of a "packaging" idea vs true fetch or load on demand - but if you're just trying to suggest something that you could do today without any new features then I guess maybe that's a start. :)

Pretty sure TIC-80 already has this stuff built in, it just doesn't expose it to carts.
